### PR TITLE
Upgrade project libs.

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

File: requirements/dev.in [hunk 1]
```diff
@@ -10,7 +10,7 @@ ipdb
 isort
 jsonschema~=2.5
 pep8
-pip==20.2.4
+pip==20.3.4
```
Comment (LeoMcA) [inline requirements/dev.in:13]: pip needs to be updated in the Dockerfile as well, but we also need to fix https://github.com/mozilla/sumo-project/issues/752 to do that.

I'd stick to bumping lxml individually for now with `pip-compile-multi --no-upgrade -g default`

**Comment by akatsoulas** [comment]
PR updated to include the fix for the referenced issue

File: requirements/default.in [hunk 2]
```diff
@@ -57,14 +57,14 @@ parameterized~=0.7.4
...
 py-wikimarkup~=2.0.1
-pyOpenSSL~=19.1.0
+pyOpenSSL~=20.0.0
 pyquery~=1.2.9
 python-dateutil~=2.8.1
 python-decouple~=3.3
 python-memcached~=1.59
 pytz~=2020.1
 redis~=3.5.2
-requests[security]~=2.23.0
```
Comment (LeoMcA) [inline requirements/default.in:67]: Was this what fixed the pip deps error?

File: requirements/dev.in [hunk 3]
```diff
@@ -10,8 +10,8 @@ ipdb
 isort
 jsonschema~=2.5
 pep8
-pip==20.2.4
-pip-compile-multi~=2.3.0
+pip~=21.0
```
Comment (LeoMcA) [inline requirements/dev.in:13]: This should be pinned to `==21.0.1` so it stays in line with the version in our docker container

**Review by LeoMcA — APPROVED** [review]
r+wc

File: requirements/default.in (hunk 2 above)
Comment (akatsoulas) [reply requirements/default.in:67]: If I understand this correctly (big if), the new resolver that was introduced in pip v20.3+, changed the way constraints work. Constraints are loaded prior to the actual resolver so when pip sees that `extra_require`, it tries to deal with it as a dependency but the hash checking mode is complaining that those extras are not pinned.

**Pull request merged**